lib/dynamoid/adapter_plugin/aws_sdk_v2.rb

```diff
@@ -38,38 +38,44 @@ def client
       #
       # @todo: Provide support for passing options to underlying batch_get_item http://docs.aws.amazon.com/sdkforruby/api/Aws/DynamoDB/Client.html#batch_get_item-instance_method
       def batch_get_item(table_ids, options = {})
-        request_items = Hash.new{|h, k| h[k] = []}
+request_items = Hash.new{|h, k| h[k] = []}
         return request_items if table_ids.all?{|k, v| v.empty?}
 
+        ret = Hash.new([].freeze) # Default for tables where no rows are returned
+
         table_ids.each do |t, ids|
           next if ids.empty?
           tbl = describe_table(t)
           hk  = tbl.hash_key.to_s
           rng = tbl.range_key.to_s
 
-          keys = if rng.present?
-            Array(ids).map do |h,r|
-              { hk => h, rng => r }
-            end
-          else
-            Array(ids).map do |id|
-              { hk => id }
+          Array(ids).each_slice(100) do |ids|
```

Comment:
All I know about batch calls is there is a limit of 25 per batch on creates and deletes.  As this is for get, not create or delete, I am not sure what the default is, or the maximum.  If Amazon has a default batch size, or a recommended batch size we should use that.  If not then 100 seems reasonable.